test/fetch.test.ts

```diff
@@ -111,120 +103,124 @@ describe('fetch', () => {
   })
 
   it(`POST /nocontent`, async () => {
-    const fun = fetcher.path('/nocontent').method('post').create()
-    const { status, data } = await fun(undefined, fetch)
+    const fun = apity.path('/nocontent').method('post').create()
+    const request = fun(undefined)
+    const { data, status } = await request.onData
     expect(status).toBe(204)
     expect(data).toBeUndefined()
   })
 
-  it('GET /error', async () => {
-    expect.assertions(3)
-
-    const fun = fetcher.path('/error/{status}').method('get').create()
-
-    try {
-      await fun({ status: 400 }, fetch)
-    } catch (err) {
-      expect(err instanceof ApiError).toBe(true)
-      expect(err instanceof fun.Error).toBe(true)
-
-      if (err instanceof ApiError) {
-        expect(err).toMatchObject({
-          status: 400,
-          statusText: 'Bad Request',
-          data: '',
-        })
-      }
-    }
-  })
-
-  it('GET /error (json body)', async () => {
-    const fun = fetcher.path('/error/{status}').method('get').create()
-
-    const errors = {
-      badRequest: false,
-      internalServer: false,
-      other: false,
-    }
-
-    const handleError = (e: any) => {
-      if (e instanceof fun.Error) {
-        const error = e.getActualType()
-        // discriminated union
-        if (error.status === 400) {
-          errors.badRequest = error.data.badRequest
-        } else if (error.status === 500) {
-          errors.internalServer = error.data.internalServer
-        } else {
-          errors.other = error.data.message === 'unknown error'
-        }
-      }
-    }
-
-    for (const status of [400, 500, 503]) {
-      try {
-        await fun({ status, detail: true }, fetch)
-      } catch (e) {
-        handleError(e)
-      }
-    }
-
-    expect(errors).toEqual({
-      badRequest: true,
-      internalServer: true,
-      other: true,
-    })
-  })
-  it('default error type {status: number, data: any}', async () => {
-    expect.assertions(2)
-
-    const fun = fetcher.path('/defaulterror').method('get').create()
-
-    try {
-      await fun({}, fetch, {})
-    } catch (e) {
-      if (e instanceof fun.Error) {
-        const error = e.getActualType()
-        expect(error.status).toBe(500)
-        expect(error.data).toEqual('internal server error')
-      }
-    }
-  })
-
-  it('network error', async () => {
-    expect.assertions(1)
-
-    const fun = fetcher.path('/networkerror').method('get').create()
-
-    try {
-      await fun({}, fetch, {})
-    } catch (e) {
-      expect(e).not.toBeInstanceOf(ApiError)
-    }
-  })
-
-  it('operation specific error type', () => {
-    const one = fetcher.path('/query/{a}/{b}').method('get').create()
-    const two = fetcher.path('/body/{id}').method('post').create()
-
-    expect(new one.Error({} as any)).not.toBeInstanceOf(two.Error)
-    expect(new two.Error({} as any)).not.toBeInstanceOf(one.Error)
-  })
+  // it('GET /error', async () => {
```

Comment:
Don't leave large commented out blocks in code. This is git, it has history and branches.